docs/guides/managing_packages.md

```diff
@@ -0,0 +1,254 @@
+During the development of agents and services, users need to manage other packages, either developing them themselves or fetching already available ones. This guide will show you which are the relevant directories during AEA development and how to get, use and publish packages.
+
+The first thing you need to know is that there are different types of packages: agents, services, connections, contracts, protocols, services and skills. These packages can live in different places:
+
+- Inside an agent if they are being used by that agent, with the exception of services which never live inside an agent.
+- In the local registry (packages folder), a local directory that stores packages classified by vendor.
+- In the remote registry, a remote machine that serves packages over IPFS or HTTP.
+
+You can see that this setup is not that different from the one git uses.
+
+<figure markdown>
+![](../images/package_management.svg){ width="85%" height="85%" style="display: block; margin: 0 auto" }
+<figcaption>Overview of the package managing flows with the Open Autonomy framework</figcaption>
+</figure>
+
+This package organization lets users develop in two different ways:
+
+- Create an agent and develop inside it. In this case you will have a vendor folder inside the agent containing all packages from other developers, and it is not recommended to modify anything inside it. Packages under development will live inside the agent (see green box on the diagram).
+
+- Develop the needed packages independently from the agent, inside the local registry (packages folder) and later add them to an agent. This is the recommended method.
+
+## The agent structure
+
+The first thing a developer needs to do before they start writing code is setting up the project. Instead of starting from scratch, we recommend to use our developer template:
+
+```bash
+git clone git@github.com:valory-xyz/dev-template.git
+cd dev-template/
+make new_env
+pipenv shell
+```
+
+The previous commands will generate a virtual environment with {{open_autonomy}} installed, an empty local registry, some useful tools for checking packages and dummy tests. After this, in the same fashion git asks to initialize its configuration using `git config`, {{open_autonomy}} asks the user to set the user name, whether we are using a local or remote registry as well as the type of registry (IPFS by default). For example, initialize your registry running the following command:
+
+```bash
+autonomy init --reset --author john_doe --remote --ipfs --ipfs-node "/dns/registry.autonolas.tech/tcp/443/https"
+```
+
+Now we are in the position of creating our first agent:
+
+```bash
+aea create my_agent
+cd my_agent
+```
+
+If you explore the newly created `my_agent` directoy, you'll see it contains the following structure:
```

Comment:
The problem here is you now first explain the not-recommended approach ...

Reply:
I know, and it was intentional as it helps to build the narrative thread. If I start from the recommended approach then the other one feels like a step-back. I also thought about deleting the non-recommended approach but when read the text felt like something was missing and the reader is left wondering why we need an intermediate, empty agent.